clients/pkg/promtail/wal/watcher.go

```diff
@@ -0,0 +1,305 @@
+package wal
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/tsdb/record"
+	"github.com/prometheus/prometheus/tsdb/wlog"
+
+	"github.com/grafana/loki/pkg/ingester/wal"
+)
+
+const (
+	readPeriod         = 10 * time.Millisecond
+	segmentCheckPeriod = 100 * time.Millisecond
+
+	// debug flag used for developing and testing the watcher code. Using this instead of level.Debug to avoid checking
+	// the log level inwards into the logger code, and just making the compiler omit this code.
+	debug = false
```

Comment:
not sure I understand the comment as to why we'd want to do this over `level.Debug`

Reply:
Good point. Thanks for the example of existing usage of the same pattern. Let keep this :+1: